Code/Source/RobotImporter/URDF/VisualsMaker.cpp

```diff
@@ -0,0 +1,209 @@
+            return;
+        }
+        auto visualEntityId = createEntityResult.GetValue();
+        AddVisualToEntity(visual, visualEntityId);
+        AddMaterialForVisual(visual, visualEntityId);
+    }
+
+    void VisualsMaker::AddVisualToEntity(urdf::VisualSharedPtr visual, AZ::EntityId entityId)
+    {
+        // Apply transform as per origin
+        PrefabMakerUtils::SetEntityTransform(visual->origin, entityId);
+
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        auto geometry = visual->geometry;
+        switch (geometry->type)
+        {
+        case urdf::Geometry::SPHERE:
+            {
+                auto sphereGeometry = std::dynamic_pointer_cast<urdf::Sphere>(geometry);
+                entity->CreateComponent(LmbrCentral::EditorSphereShapeComponentTypeId);
+                entity->Activate();
+                LmbrCentral::SphereShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::SphereShapeComponentRequests::SetRadius, sphereGeometry->radius);
+                entity->Deactivate();
+            }
+            break;
+        case urdf::Geometry::CYLINDER:
+            {
+                auto cylinderGeometry = std::dynamic_pointer_cast<urdf::Cylinder>(geometry);
+                entity->CreateComponent(LmbrCentral::EditorCylinderShapeComponentTypeId);
+                entity->Activate();
+                LmbrCentral::CylinderShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::CylinderShapeComponentRequests::SetRadius, cylinderGeometry->radius);
+                LmbrCentral::CylinderShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::CylinderShapeComponentRequests::SetHeight, cylinderGeometry->length);
+                entity->Deactivate();
+            }
+            break;
+        case urdf::Geometry::BOX:
+            {
+                auto boxGeometry = std::dynamic_pointer_cast<urdf::Box>(geometry);
+                entity->CreateComponent(LmbrCentral::EditorBoxShapeComponentTypeId);
+                AZ::Vector3 boxDimensions = URDF::TypeConversions::ConvertVector3(boxGeometry->dim);
+                entity->Activate();
+                LmbrCentral::BoxShapeComponentRequestsBus::Event(
+                    entityId, &LmbrCentral::BoxShapeComponentRequests::SetBoxDimensions, boxDimensions);
+                entity->Deactivate();
+            }
+            break;
+        case urdf::Geometry::MESH:
+            {
+                auto meshGeometry = std::dynamic_pointer_cast<urdf::Mesh>(geometry);
+
+                // TODO - a PoC solution for path, replace with something generic, robust, proper
+                std::filesystem::path modelPath(m_modelPath.c_str());
+                modelPath = modelPath.remove_filename();
+                auto relativePathToMesh = std::regex_replace(meshGeometry->filename, std::regex("package://"), "");
+                modelPath += relativePathToMesh;
```

Comment:
The reason I recommended against using the `std::filesystem::path` functions is that it exposes what I believe is a dangerous [string concatenation operator](https://en.cppreference.com/w/cpp/filesystem/path/concat).

This code here is joining two paths together smartly with a path separator in between but instead appending two strings.
You would want to use the `operator/=` here, not `operator+=`.
The AZ::IO::Path API omits the plus('+') operator to avoid that danger.

```suggestion
                AZ::IO::Path modelPath(m_modelPath);
                modelPath().RemoveFilename(); 
                AZ::IO::Path meshFilename(AZStd::string_view(meshGeometry->filename.c_str(), meshGeometry->filename.size()));
                AZ::IO::StringFunc::Replace(meshFilename.Native(), "package://", "", true, true);
                modelPath /= meshFilename;
```